src/cli/self_update.rs

```diff
@@ -197,46 +148,135 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         }
     }
 
-    eprintln!(
-        "{}Pixi archive downloaded.",
-        console::style(console::Emoji("✔ ", "")).green(),
-    );
+    Ok(archived_tempfile)
+}
 
+/// Unpacks a pixi archive (typically downloaded from GitHub releases) into a temporary directory.
+async fn unpack_release_archive(
+    mut archived_tempfile: NamedTempFile,
+    archive_name: &str,
+) -> miette::Result<TempDir> {
     // Seek to the beginning of the file before uncompressing it
     let _ = archived_tempfile.rewind();
 
     // Create a temporary directory to unpack the archive
-    let binary_tempdir = &tempfile::tempdir().into_diagnostic()?;
+    let binary_tempdir = tempfile::tempdir().into_diagnostic()?;
 
     // Uncompress the archive
     if archive_name.ends_with(".tar.gz") {
-        unpack_tar_gz(&archived_tempfile, binary_tempdir)?;
+        unpack_tar_gz(&archived_tempfile, &binary_tempdir)?;
     } else if archive_name.ends_with(".zip") {
         let mut archive = zip::ZipArchive::new(archived_tempfile.as_file()).into_diagnostic()?;
-        archive.extract(binary_tempdir).into_diagnostic()?;
+        archive.extract(&binary_tempdir).into_diagnostic()?;
     } else {
         let error_message = format!("Unsupported archive format: {}", archive_name);
         Err(miette::miette!(error_message))?
     }
 
-    eprintln!(
-        "{}Pixi archive uncompressed.",
-        console::style(console::Emoji("✔ ", "")).green(),
-    );
+    Ok(binary_tempdir)
+}
 
-    // Get the new binary path used for self-replacement
-    let new_binary_path = binary_tempdir.path().join(pixi_binary_name());
+pub async fn execute(args: Args) -> miette::Result<()> {
+    if let Some(url) = args.url {
+        // If a URL is provided, download the archive from the URL
+        let new_binary_path = download(url).await?;
```

Comment:
should we also provide a UX message like `eprintln!("Downloading pixi")`

Reply:
Sure!

Reply:
Any preference for the styling, like an emoji or a specific styling to apply to those kind of "doing something" messages?